### PR TITLE
Improved error handling for email support

### PR DESCRIPTION
**Pull request by SchrodingersGat**
- Prevent email sending if email not configured
- Check for tx email address before sending

(cherry picked from commit de541f811ede030ea5eb3136132731e1dafccc31)

Ref: https://github.com/inventree/InvenTree/issues/4769

### CODE REVIEW

File: InvenTree/email.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        # Display warning unless in test mode
+        if not settings.TESTING:  # pragma: no cover
+            logger.debug("EMAIL_HOST is not configured")
+
+    # Display warning unless in test mode
+    if not settings.TESTING:  # pragma: no cover
+        logger.debug("EMAIL_HOST_USER is not configured")
+
+    # Display warning unless in test mode
+    if not settings.TESTING:  # pragma: no cover
```
Comment (matmair) [inline InvenTree/email.py:39]: Question: Where is the check for user or password?

**Review by matmair — APPROVED** [review]

File: InvenTree/email.py (hunk 1 above)
Comment (SchrodingersGat) [reply InvenTree/email.py:39]: https://github.com/inventree/InvenTree/pull/4492

Comment (matmair) [reply InvenTree/email.py:39]: Seems like an error in the initial PR by me. We should check before logging - not error out though

Comment (SchrodingersGat) [reply InvenTree/email.py:39]: @matmair I'm not 100% across what you're suggesting here - want to make a suggested change?

File: InvenTree/email.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    if not settings.EMAIL_HOST:
+        configured = False
+
+        # Display warning unless in test mode
+        if not settings.TESTING:  # pragma: no cover
+            logger.debug("EMAIL_HOST is not configured")
+
+    # Display warning unless in test mode
+    if not settings.TESTING:  # pragma: no cover
+        logger.debug("EMAIL_HOST_USER is not configured")
```
Comment (matmair) [inline InvenTree/email.py:36]:
```suggestion
    if not settings.EMAIL_HOST_USER and not settings.TESTING:  # pragma: no cover
        logger.debug("EMAIL_HOST_USER is not configured")
```

File: InvenTree/email.py (hunk 1 above)
Comment (matmair) [inline InvenTree/email.py:39]:
```suggestion
    if not settings.EMAIL_HOST_PASSWORD and not settings.TESTING:  # pragma: no cover
```

**Review by matmair — CHANGES_REQUESTED** [review]

**Review by matmair — APPROVED** [review]

**Pull request merged**